### PR TITLE
Promote existing E2Es for hostport/HostIP resolution to conformance

### PR DESCRIPTION
**Pull request by globervinodhn**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
/kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**: This PR is a request to promote existing E2Es for service resolution with respect to HostIP and HostPort to conformance.

**Which issue(s) this PR fixes**: NONE. Requests for promotion of existing E2Es
Fixes #

**Special notes for your reviewer**: The two existing E2Es that this PR requests to promote and their relevant information is as provided below,
1. "validates that there is no conflict between pods with same hostPort but different hostIP and protocol"
         --- Takes approximately 45 seconds for successful execution.
         --- The test MUST pass as HostIP resolves to a kubernetes node and usage of the same port in
         different nodes should not be a conflict scenario.
2. "validates that there exists conflict between pods with same hostPort and protocol but one using 0.0.0.0 hostIP"
         --- Takes approximately 320 seconds for successful execution.

Both the tests are found to be non-flaky and non-disruptive.

**Does this PR introduce a user-facing change?**: NONE

release-note-none

/area conformance
@mgdevstack @brahmaroutu

### CODE REVIEW

File: test/e2e/scheduling/predicates.go [hunk 1]
```diff
@@ -554,7 +554,14 @@ var _ = SIGDescribe("SchedulerPredicates [Serial]", func() {
 		verifyResult(cs, 1, 0, ns)
 	})
 
-	ginkgo.It("validates that there is no conflict between pods with same hostPort but different hostIP and protocol", func() {
+	/*
+		Release : v1.15
+		Testname: Services, HostPort matching and HostIP and Protocol not-matching
```
Comment (johnbelamaric) [inline test/e2e/scheduling/predicates.go]: This test only tests the scheduler and how it handles host ports. It has nothing to do with services. The description is also incorrect. These pods are scheduled to the same node. A better name and description would be:
```
Testname: Scheduling, HostPort matching but HostIP or Protocol is different
Description: Pods with the same HostPort value MUST be able to be scheduled to the same node if the HostIP or Protocol is different.
```

File: test/e2e/scheduling/predicates.go [hunk 2]
```diff
@@ -581,7 +588,13 @@ var _ = SIGDescribe("SchedulerPredicates [Serial]", func() {
 		createHostPortPodOnNode(f, "pod3", ns, "127.0.0.2", port, v1.ProtocolUDP, nodeSelector, true)
 	})
 
-	ginkgo.It("validates that there exists conflict between pods with same hostPort and protocol but one using 0.0.0.0 hostIP", func() {
+	/*
+		Release : v1.15
+		Testname: Services, HostPort and Protocol matching and default HostIP (0.0.0.0)
```
Comment (johnbelamaric) [inline test/e2e/scheduling/predicates.go]: Similarly, these are incorrect. Better:

```
Testname: Scheduling, HostPort and Protocol match, HostIPs different but one is default HostIP (0.0.0.0)
Description: Pods with the same HostPort and Protocol, but different HostIPs, MUST NOT schedule to the same node if one of those IPs is the default HostIP of 0.0.0.0, which represents all IPs on the host.
```

File: test/e2e/scheduling/predicates.go [hunk 3]
```diff
@@ -554,7 +554,14 @@ var _ = SIGDescribe("SchedulerPredicates [Serial]", func() {
 		verifyResult(cs, 1, 0, ns)
 	})
 
-	ginkgo.It("validates that there is no conflict between pods with same hostPort but different hostIP and protocol", func() {
+	/*
+		Release : v1.15
```
Comment (johnbelamaric) [inline test/e2e/scheduling/predicates.go]: At this point these should be 1.16.

**Review by johnbelamaric — CHANGES_REQUESTED** [review]

**Review by bsalamat — COMMENTED** [review]
/lgtm
/approve

**Comment by bgrant0607** [comment]
For reference, the fields whose scheduling behavior are being exercised are documented here:
https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.15/#containerport-v1-core

Thanks!

/milestone v1.16
/lgtm
/approve

File: test/conformance/testdata/conformance.txt [hunk 4]
```diff
@@ -206,6 +206,8 @@ test/e2e/node/pre_stop.go: "should call prestop when killing a pod"
 test/e2e/scheduling/predicates.go: "validates resource limits of pods that are allowed to run"
 test/e2e/scheduling/predicates.go: "validates that NodeSelector is respected if not matching"
 test/e2e/scheduling/predicates.go: "validates that NodeSelector is respected if matching"
+test/e2e/scheduling/predicates.go: "validates that there is no conflict between pods with same hostPort but different hostIP and protocol"
+test/e2e/scheduling/predicates.go: "validates that there exists conflict between pods with same hostPort and protocol but one using 0.0.0.0 hostIP"
```
Comment (BenTheElder) [inline test/conformance/testdata/conformance.txt]: :/ we should please not use IPv4 only, Kubernetes should be able to be IPv6 as well. cc @aojea

Comment (aojea) [reply test/conformance/testdata/conformance.txt]: I´m working on it, the test is fine, is passing an empty string, but the function I added to translate IPv4 to IPv6 addresses is not checking for that, creating a non-valid IPv6 address
My fault

Comment (BenTheElder) [reply test/conformance/testdata/conformance.txt]: We should also rephrase the test name IMHO.
Thanks for working on this.

**Pull request merged**